### PR TITLE
[1.10] Bump curl 7.83.0

### PR DESCRIPTION
**Pull request by ligurio**

### CODE REVIEW

File: changelogs/unreleased/bump-libcurl-to-7.83.0.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## feature/build
+
+* Bump Curl from 7.80.0 to 7.83.0 for static build (gh-6029).
```
Comment (Totktonada) [inline changelogs/unreleased/bump-libcurl-to-7.83.0.md:3]: The same comments as in #7119 are applicable here.

File: changelogs/unreleased/bump-libcurl-to-7.83.0.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## feature/build
```
Comment (Totktonada) [inline changelogs/unreleased/bump-libcurl-to-7.83.0.md:1]: Shouldn't we update configure options as we do for CMake in #7119?

File: changelogs/unreleased/bump-libcurl-to-7.83.0.md (hunk 1 above)
Comment (ligurio) [reply changelogs/unreleased/bump-libcurl-to-7.83.0.md:3]: Updated.

File: changelogs/unreleased/bump-libcurl-to-7.83.0.md (hunk 2 above)
Comment (ligurio) [reply changelogs/unreleased/bump-libcurl-to-7.83.0.md:1]: added

File: cmake/BuildLibCURL.cmake [hunk 3]
```diff
@@ -149,6 +150,15 @@ macro(curl_build)
                 --disable-progress-meter
                 --disable-dnsshuffle
                 --disable-alt-svc
+                --disable-hsts
```
Comment (Totktonada) [inline cmake/BuildLibCURL.cmake:153]: HSTS is enabled in #7119 (and I don't see a reason to disable it if it does not engage extra dependencies).

**Review by Totktonada — APPROVED** [review]
I can't spot problems visually. As far as I can judge, the changes are okay.

File: cmake/BuildLibCURL.cmake (hunk 3 above)
Comment (ligurio) [reply cmake/BuildLibCURL.cmake:153]:
```diff
--- a/cmake/BuildLibCURL.cmake
+++ b/cmake/BuildLibCURL.cmake
@@ -150,7 +150,6 @@ macro(curl_build)
                 --disable-progress-meter
                 --disable-dnsshuffle
                 --disable-alt-svc
-                --disable-hsts
                 --enable-libcurl-option
                 --enable-openssl-auto-load-config
                 --enable-socketpair
```

Comment (Totktonada) [reply cmake/BuildLibCURL.cmake]: We agreed to list all options explicitly, but here you just removed it.

AFAIR, alt-svc and dnsshuffle also different between branches.

Comment (ligurio) [reply cmake/BuildLibCURL.cmake]: > We agreed to list all options explicitly, but here you just removed it.

Fixed.

> AFAIR, alt-svc and dnsshuffle also different between branches.

Fixed.

**Pull request merged**